### DISCUSSION TITLE
ModuleNotFoundError on vercel but working on Local

### DISCUSSION THREAD

**Discussion by allindeveloper** [vercel/next.js#16179] [Help]
# Bug report

## Describe the bug

A get a ModuleNotFoundError when deploying my application to vercel.com but it works fine on my local

01:51:17 | Failed to compile.
-- | --
01:51:17 | ModuleNotFoundError: Module not found: Error: Can't resolve 'common/src/theme/agency' in '/vercel/6965fa7d/pages'
01:51:17 | > Build error occurred
01:51:17 | Error: > Build failed because of webpack errors
01:51:17 | at build (/vercel/6965fa7d/node_modules/next/dist/build/index.js:15:918)
01:51:17 | at processTicksAndRejections

## Expected behavior

I expect that the build on vercel should be fine

## System information

- OS: [Windows]
- Browser (if applies) [chrome]
- Version of Next.js: [e.g. 9.5.0]
- Version of Node.js: [e.g. 12.18.2]

## Additional context

Package.json file
{
  "name": "next-landing",
  "version": "6.0.0",
  "private": true,
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start",
    "export": "next build && next export",
    "now-build": "next build && next export -o dist",
    "netlify-build": "next build && next export"
  },
  
  "dependencies": {
    "@glidejs/glide": "^3.4.1",
    "@reach/accordion": "^0.10.4",
    "@redq/reuse-modal": "^1.1.9",
    "@styled-system/theme-get": "^5.1.2",
    "@zeit/next-css": "^1.0.1",
    "firebase-admin": "^8.12.1",
    "firebase-functions": "^3.7.0",
    "next": "9.5.0",
    "next-compose-plugins": "^2.2.0",
    "next-fonts": "1.0.3",
    "next-optimized-images": "^2.6.1",
    "next-transpile-modules": "4.0.0",
    "prop-types": "^15.7.2",
    "rc-drawer": "^4.1.0",
    "rc-progress": "^3.0.0",
    "rc-tabs": "9.7.0",
    "react": "^16.13.1",
    "react-accessible-accordion": "3.0.1",
    "react-anchor-link-smooth-scroll": "^1.0.12",
    "react-collapser": "^1.5.10",
    "react-content-loader": "^5.1.0",
    "react-countdown-now": "^2.1.2",
    "react-countup": "^4.3.3",
    "react-dom": "^16.13.1",
    "react-icons-kit": "^1.3.1",
    "react-id-swiper": "^4.0.0",
    "react-image": "^4.0.1",
    "react-image-gallery": "0.8.18",
    "react-masonry-component": "^6.2.1",
    "react-particles-js": "^3.2.1",
    "react-reveal": "^1.2.2",
    "react-rnd": "^10.1.10",
    "react-scroll-parallax": "^2.3.3",
    "react-scrollspy": "^3.4.2",
    "react-select": "^3.1.0",
    "react-stickynode": "2.1.1",
    "styled-components": "^5.1.1",
    "styled-system": "5.1.5",
    "swiper": "^5.4.5"
  },
  "devDependencies": {
    "babel-plugin-styled-components": "^1.10.7",
    "imagemin-mozjpeg": "^9.0.0",
    "imagemin-optipng": "^8.0.0",
    "imagemin-svgo": "^8.0.0"
  },
  "engines": {
    "node": "12.x"
  },
  "module": "pages/index.js"
}

Any help Please..

**Comment by Timer** [comment]
Please double check that the casing of your file import matches what's on disk. Windows isn't case sensitive like our Linux build environments are!

For example, your `agency.js` might actually be `Agency.js`, thus would require:
```js
import 'common/src/theme/Agency'
```

**Comment by allindeveloper** [comment]
I had to change the folder name, still did not work.

06:57:06 | Failed to compile.
-- | --
06:57:06 | ModuleNotFoundError: Module not found: Error: Can't resolve 'common/src/theme/agencynew' in '/vercel/51d6c854/pages'
06:57:06 | > Build error occurred
06:57:06 | Error: > Build failed because of webpack errors
06:57:06 | at build (/vercel/51d6c854/node_modules/next/dist/build/index.js:15:918)
06:57:06 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
06:57:06 | error Command failed with exit code 1.

**Comment by allindeveloper** [comment]
![image](https://user-images.githubusercontent.com/20103353/90218634-35109e00-ddfc-11ea-9d18-9134a5e1e9c0.png)


This the pages/index == >

`import  agencytheme from 'common/src/theme/agencynew';`

`<ThemeProvider theme={agencytheme}>
</ThemeProvider>`

**Comment by allindeveloper** [comment]
@Timer 
So i observed that running `now` in my code editor pushed the code to vercel and that works pretty well

where as

when i do a github integration.. this get the `ModuleNotFoundError`

Project Setup
![image](https://user-images.githubusercontent.com/20103353/90221727-a18e9b80-de02-11ea-8469-fdd8a85c63f2.png)

`the main application is within `packages/landing`


How can i do a github integration without the ModuleNotFoundError

is it because of the now.json configuration that i have ??

**Reply by timneutkens** [reply]
> when i do a github integration.. this get the ModuleNotFoundError

It's likely that you changed the casing but didn't update the reference of it in git, when renaming try using `git mv OldFile.js oldfile.js` (changing casing)

**Reply by ailtonbueno** [reply]
Perfectly. after change casing name of file or folder. I did it and solve may problem.

**Reply by dellwatson** [reply]
weird, same things happen to me. I just move the folder to trigger fresh build and it works

**Reply by jayantasamaddar** [reply]
I've faced this error but resolved it. I was in a monorepo and my Next.js build script had a certain dependency from one of the monorepo packages. Hence the package had to be build first before the script even ran. Once I handled that, it deployed smooth.

**Comment by emilienbidet** [comment]
This works for me :
- Removing conflicting files from repo 
- Adding them again

As wrote in [this Vercel docs](https://vercel.com/guides/how-do-i-resolve-a-module-not-found-error), make sure git don't ignore the case, you can set this option :
`git config core.ignorecase false`

**Reply by malimalik** [reply]
Thanks using `git config core.ignorecase false` actually worked.
